cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogInspectCommand.java

```diff
@@ -0,0 +1,369 @@
+                    ((DeleteSegmentOperation) op).getStreamSegmentId(), ((DeleteSegmentOperation) op).getStreamSegmentOffset(),
+                    OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        } else if (op instanceof MetadataCheckpointOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        } else if (op instanceof StorageMetadataCheckpointOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        } else if (op instanceof StreamSegmentMapOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    ((StreamSegmentMapOperation) op).getStreamSegmentId(), OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        }
+        return res;
+    }
+
+
+    private int filterResult(Predicate<OperationInspectInfo> predicate, int containerId, DebugDurableDataLogWrapper originalDataLog) throws Exception {
+        AtomicInteger res = new AtomicInteger();
+
+        @Cleanup
+        FileWriter writer = new FileWriter(createFileAndDirectory(getCLIControllerConfig().getInspectResultLocation()));
+
+        readDurableDataLogWithCustomCallback((a, b) -> {
+                    if (predicate.test(getActualOperation(a))) {
+                        output(a.toString());
+                        try {
+                            writer.write(getActualOperation(a).toString() + NEW_LINE_SEPARATOR);
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                        res.getAndIncrement();
+                    }
+                }, containerId, originalDataLog.asReadOnly());
+        writer.flush();
+        writer.close();
+        return res.get();
+    }
+
+    /**
+     * Reads a {@link DurableDataLog} associated with a container id and runs the callback on each {@link Operation}
+     * read from the log.
+     *
+     * @param callback Callback to be run upon each {@link Operation} read.
+     * @param containerId Container id to read from.
+     * @param durableDataLog {@link DurableDataLog} of the Container to be read.
+     * @return Number of {@link Operation}s read.
+     * @throws Exception If there is a problem reading the {@link DurableDataLog}.
+     */
+    @VisibleForTesting
+    int readDurableDataLogWithCustomCallback(BiConsumer<Operation, List<DataFrameRecord.EntryInfo>> callback,
```

Comment:
This is the same method already defined in `DurableDataLogRepairCommand`. Do not copy methods over again if not needed, just reuse them via inheritance.

Reply:
corrected and reusing the same